compiler/base.ts

```diff
@@ -133,6 +242,14 @@ export class LineCompiler implements InstrVisitor<void>, ExprVisitor<void> {
         }
       }
 
+      try {
+        this.checkBlocksClosed();
+      } catch (err) {
+        if (!(err instanceof Synchronize)) {
```

Comment:
Should this try/catch be pushed into checkBlocksClosed?

Reply:
I'm going to say "no", because `checkBlocksClosed` doesn't have any way of enforcing that it's being called at the end of parsing. `compile()` is the method that ensures that, and hence can make that decision.